File: LowLevelDesign/tic-tac-toe/src/strategy/RowWinningStrategy.ts
import { Board } from "../Board";
import { Player } from "../Player";
import { WinningStrategy } from "./WinningStrategy";

export class RowWinningStrategy  implements WinningStrategy {
    checkWinner(board: Board, player: Player): boolean {
        for(let row = 0; row < board.size; row++) {
            let rowWin = true;
            for(let col = 0; col < board.size; col++) {
                if(board.board[row][col].getSymbol() !== player.symbol) {
                    rowWin = false;
                    break;
                }
            }
            if(rowWin) return true;
        }
        return false
    }
}

File: SystemDesignDiagrams/practice/LLD/tictactoe/src/main/java/com/tictactoe/strategies/RowWinningStrategy.java
package com.tictactoe.strategies;

import com.tictactoe.entities.Board;
import com.tictactoe.entities.Game;
import com.tictactoe.entities.Player;

public class RowWinningStrategy implements WinningStrategy {
    public boolean checkWinner(Game game) {
        Board board = game.getBoard();
        Player currentPlayer = game.getCurrentPlayer();
            for(int i=0;i<board.getSize();i++) {
                boolean rowWin = true;
                for(int j=0;j<board.getSize();j++) {
                    if(board.getBoard()[i][j].getSymbol() != currentPlayer.getSymbol()) {
                        rowWin = false;
                        break;
                    }
                }
                if(rowWin) {
                    return true;
                }
            }


        return false;
    }
}
